### PR TITLE
Implement repeated tasks in FakeScheduledExecutorService

### PR DESCRIPTION
**Pull request by keeferrourke**
Implemented the  `scheduleWithFixedDelay` and `scheduleWithFixedRate` functions for the _FakeScheduledExecutorService_.

Since this is a single-threaded executor,  and the service always has to catch up to the clock, fixed delay is no different from fixed rate to my understanding.

### CODE REVIEW

File: misk-testing/src/main/kotlin/misk/concurrent/FakeScheduledExecutorService.kt [hunk 1]
```diff
@@ -46,39 +72,80 @@ class FakeScheduledExecutorService @Inject constructor(private val clock: Clock)
...
+    val isRepeated: Boolean get() = fixedDelay > 0
+
+    override fun run() {
+      if (isCancelled) throw CancellationException("FutureTask was cancelled")
+      super.run()
+    }
+
+    public override fun runAndReset(): Boolean {
+      return super.runAndReset()
+    }
```
Comment (mightyguava) [inline misk-testing/src/main/kotlin/misk/concurrent/FakeScheduledExecutorService.kt]: This seems superfluous?

File: misk-testing/src/main/kotlin/misk/concurrent/FakeScheduledExecutorService.kt [hunk 2]
```diff
@@ -17,21 +17,47 @@ import javax.inject.Singleton
...
 
   /** Check the current time on the clock and run any scheduled tasks that are due. */
   fun tick() {
-    futures.entries().stream()
-        .filter { entry -> entry.key <= clock.millis() }
-        .forEach { entry -> entry.value.run() }
+    var future = futures.poll()
+    while (future != null) {
+      if (this.isShutdown) {
+        future.cancel(false)
```
Comment (mightyguava) [inline misk-testing/src/main/kotlin/misk/concurrent/FakeScheduledExecutorService.kt]: Why did you need to add cancellation? If tick isn't called, no futures will run.

File: misk-testing/src/main/kotlin/misk/concurrent/FakeScheduledExecutorService.kt [hunk 3]
```diff
@@ -46,39 +72,80 @@ class FakeScheduledExecutorService @Inject constructor(private val clock: Clock)
...
+      if (isCancelled) throw CancellationException("FutureTask was cancelled")
+      super.run()
+    }
+
+    public override fun runAndReset(): Boolean {
+      return super.runAndReset()
+    }
+
+    internal fun reschedule() {
+      if (!isRepeated) return
```
Comment (mightyguava) [inline misk-testing/src/main/kotlin/misk/concurrent/FakeScheduledExecutorService.kt]: redundant check. Might be better as an assert.

File: misk-testing/src/main/kotlin/misk/concurrent/FakeScheduledExecutorService.kt (hunk 1 above)
Comment (keeferrourke) [reply misk-testing/src/main/kotlin/misk/concurrent/FakeScheduledExecutorService.kt]: I would agree, but unfortunately it seems that `runAndReset()` needs to be overridden to make it a public method, so it may be used by the enclosing class.

File: misk-testing/src/main/kotlin/misk/concurrent/FakeScheduledExecutorService.kt (hunk 2 above)
Comment (keeferrourke) [reply misk-testing/src/main/kotlin/misk/concurrent/FakeScheduledExecutorService.kt]: Good point! Removed the cancellation.

File: misk-testing/src/main/kotlin/misk/concurrent/FakeScheduledExecutorService.kt (hunk 3 above)
Comment (keeferrourke) [reply misk-testing/src/main/kotlin/misk/concurrent/FakeScheduledExecutorService.kt]: Changed this to a `check` call.
